replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/HeartbeaterAndVerifierImpl.java

```diff
@@ -73,23 +135,307 @@ public void cleanUp() {
+              // status will just be overridden for current running jobs
+              replicationStatus.setStatus(Status.CONNECTION_LOST);
+            } else {
+              replicationStatus.setStatus(Status.CONNECTION_UNAVAILABLE);
+            }
+          });
+    } else {
+      changeConfigStatusFor(
+          site,
+          (replicationStatus, status) -> {
+            if (status == Status.CONNECTION_LOST || status == Status.CONNECTION_UNAVAILABLE) {
+              replicationStatus.setStatus(Status.PENDING);
+            }
+          });
+    }
+  }
+
+  private void changeConfigStatusFor(
+      ReplicationSite site, BiConsumer<ReplicationStatus, Status> updater) {
+    configManager
+        .objects()
+        .filter(config -> config.sourceOrDestinationIs(site.getId()))
+        .forEach(
+            config -> {
+              Optional<ReplicationStatus> statusOptional = statusFor(config);
+              if (statusOptional.isPresent()) {
+                ReplicationStatus replicationStatus = statusOptional.get();
+                updater.accept(replicationStatus, replicationStatus.getStatus());
+
+                try {
+                  // todo this is going to undesirably condense the status' stats...
+                  history.addReplicationEvent(replicationStatus);
+                } catch (ReplicationPersistenceException e) {
+                  LOGGER.debug(
+                      "Failed to update status while heartbeating site {}", site.getUrl(), e);
+                }
+              } else {
+                LOGGER.warn("No existing status to update for config {}", config.getName());
+              }
+            });
+  }
+
+  private Optional<ReplicationStatus> statusFor(ReplicatorConfig config) {
+    return history.getReplicationEvents(config.getName()).stream().findFirst();
+  }
+
+  @SuppressWarnings("squid:S2189" /* Run inside executor and will exit loop on executor shutdown */)
+  private void processingLoop() {
+    while (true) {
+      try {
+        processNextSite();
+      } catch (InterruptedException e) { // propagate interruption and bail
+        LOGGER.trace("Heartbeater was interrupted");
+        Thread.currentThread().interrupt();
+        return;
+      }
+    }
+  }
+
+  private DiscoveryApi createDiscoveryClient(final String url) {
+    final ObjectMapper objectMapper = new ObjectMapper();
+    objectMapper.findAndRegisterModules();
+    objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+    SecureCxfClientFactory<DiscoveryApi> cxfFactory =
+        clientFactoryFactory.getSecureCxfClientFactory(
+            url,
+            DiscoveryApi.class,
+            Collections.singletonList(
+                new JacksonJaxbJsonProvider(
+                    objectMapper, JacksonJaxbJsonProvider.DEFAULT_ANNOTATIONS)),
+            null,
+            true /* disableCnCheck */,
+            false /* allowRedirects */,
+            30000 /* connectionTimeout */,
```

Comment:
❓ Should these timeouts be constants or better yet system properties?
❓ Is 30 seconds not a long time to wait?

Reply:
30 seconds is typically what we use (and is the default of the CxfClientFactory) due to limitations of the environment.
I'd prefer not to make more configuration for these, but if we feel we need to we can.

Reply:
Not knowing what will happen on certain networks, I would feel better if we had a way to have our customer tweaks these as opposed to delivering a code patch. We can read system properties while defaulting in code to 30 secs.